deployment/terraform/agent.go

```diff
@@ -85,83 +87,118 @@ func (t *Terraform) configureAndRunAgents(extAgent *ssh.ExtAgent) error {
 		}
 	}
 
+	wg := sync.WaitGroup{}
+	var foundErr atomic.Bool
+
 	for i, val := range t.output.Agents {
-		sshc, err := extAgent.NewClient(val.PublicIP)
-		if err != nil {
-			return err
-		}
-		mlog.Info("Configuring agent", mlog.String("ip", val.PublicIP))
-		if uploadBinary {
-			dstFilePath := "/home/ubuntu/tmp.tar.gz"
-			mlog.Info("Uploading binary", mlog.String("file", packagePath))
-			if out, err := sshc.UploadFile(packagePath, dstFilePath, false); err != nil {
-				return fmt.Errorf("error uploading file %q, output: %q: %w", packagePath, out, err)
-			}
-		}
+		wg.Add(1)
+		agentNumber := i
+		instance := val
 
-		cmd := strings.Join(commands, " && ")
-		if out, err := sshc.RunCommand(cmd); err != nil {
-			return fmt.Errorf("error running command, got output: %q: %w", out, err)
-		}
+		go func() {
+			// Copy val into instance
+			defer wg.Done()
```

Comment:
Why is the `// Copy val into instance` comment inside the goroutine?

Reply:
Probably I moved some lines around during development. Let me remove it.